packages/visual-layer/src/layers/bar/bar-helper.js

```diff
@@ -183,27 +177,36 @@ export const getTranslatedPoints = (context, data, sizeConfig) => {
             sizeEncoding
         }, axes);
 
-        const { color, rawColor } = getLayerColor({ datum: d, index: i },
-            { colorEncoding, colorAxis, colorFieldIndex });
-
-        style.fill = color;
-        meta.stateColor = {};
-        meta.originalColor = rawColor;
-        meta.colorTransform = {};
+        let color = colorAxis.getColor(d.color);
 
         const update = dimensions.update;
-
-        if (!isNaN(update.x) && !isNaN(update.y) && d._id !== undefined) {
+        const resolvedEncodings = resolveEncodingValues({
+            values: {
+                x: update.x,
+                y: update.y,
+                width: update.width,
+                height: update.height,
+                color
+            },
+            data: d
+        }, i, data, context);
+        color = resolvedEncodings.color;
+        if (!isNaN(update.x) && !isNaN(update.y) && d.rowId !== undefined) {
```

Comment:
update.x and update.y are referenced multiple times. can be stored in a variable, same for d.rowId